### PR TITLE
GDB-10021 - Add Ontop create and edit tests

### PR DESCRIPTION
**Pull request by DesiBorisova**
## What? 
New tests added for the creation and edit of Ontop repositories.

## Why?
The idea is to cover a few more cases in order to avoid missing bugs when creating and editing Ontop repositories.

## How?
Tests added.

### CODE REVIEW

File: test-cypress/integration/repository/ontop-repository.spec.js [hunk 1]
```diff
@@ -152,6 +239,131 @@ describe('Ontop repositories', () => {
...
+                "type": "ontop",
+                "sesameType": "graphdb:OntopRepository",
+                "location": "",
+                "hostName": "localhost",
+                "params": { ...commonRepositoryParams, ...extraParams }
+            }
+        });
+    }
+
+    function stubRepoCreationEndpoints() {
```
Comment (svilenvelikov) [inline test-cypress/integration/repository/ontop-repository.spec.js:309]: All stubs needs to be in respective classes here https://github.com/Ontotext-AD/graphdb-workbench/tree/master/test-cypress/stubs

File: test-cypress/integration/repository/ontop-repository.spec.js [hunk 2]
```diff
@@ -152,6 +239,131 @@ describe('Ontop repositories', () => {
...
+                        "readable": true,
+                        "writable": true,
+                        "unsupported": false,
+                        "state": "ACTIVE"
+                    }
+                ]
+            }
+        }).as('getMockRepositories');
+
+        cy.intercept(`rest/repositories/${repositoryId}/restart?location=*`, {
```
Comment (svilenvelikov) [inline test-cypress/integration/repository/ontop-repository.spec.js:332]: Also try to implement the stubs in their respective classes so that we can reuse and compose them.

File: test-cypress/integration/repository/ontop-repository.spec.js [hunk 3]
```diff
@@ -152,6 +239,131 @@ describe('Ontop repositories', () => {
...
     });
+
+    function assertRepositoryStatus(repositoryId, status) {
+        cy.waitUntil(() =>
+            RepositorySteps.getRepositoryFromList(repositoryId)
+                .should('be.visible')
+                .find('.repository-status .text-secondary').should('contain.text', status));
+    }
+
+    const commonRepositoryParams = {
```
Comment (svilenvelikov) [inline test-cypress/integration/repository/ontop-repository.spec.js:251]: What about moving this in a fixture file?

File: test-cypress/integration/repository/ontop-repository.spec.js [hunk 4]
```diff
@@ -152,6 +239,131 @@ describe('Ontop repositories', () => {
 
         // Then I expect url to be calculated properly
         OntopRepositorySteps.getUrlInput().should('have.value', 'jdbc:snowflake://someHostName.snowflakecomputing.com:1234/?warehouse=test_database');
-
     });
+
+    function assertRepositoryStatus(repositoryId, status) {
```
Comment (svilenvelikov) [inline test-cypress/integration/repository/ontop-repository.spec.js:244]: Why is this here and not in some steps or command file? Did you check if there is no such validation already?

File: test-cypress/integration/repository/ontop-repository.spec.js [hunk 5]
```diff
@@ -137,6 +139,91 @@ describe('Ontop repositories', () => {
...
     });
 
+    it('should create ontop repository', () => {
+        stubRepoCreationEndpoints();
+        // When I visit the create ontop page
+        OntopRepositorySteps.visitCreate();
+        // The test connection button should be disabled
+        OntopRepositorySteps.getTestConnectionButton().should('be.disabled');
+        // Then I fill the repository ID
+        RepositorySteps.getRepositoryIdField().type(repositoryId);
```
Comment (svilenvelikov) [inline test-cypress/integration/repository/ontop-repository.spec.js:149]: Implement methods like `writeRepositoryId(repositoryId)` in the steps file. All low level interactions with the web page must be in the steps.

File: test-cypress/steps/repository-steps.js [hunk 6]
```diff
@@ -121,6 +121,26 @@ export class RepositorySteps {
...
+
+    static typeUsernameInEditRepo(username) {
+        this.getUsernameFieldEditRepo().type(username);
+    }
+
+    static clickSaveEditedRepo() {
+        cy.get('#addEditOntopRepository').click();
+    }
+
+    static clickModalOK() {
```
Comment (svilenvelikov) [inline test-cypress/steps/repository-steps.js:136]: There is a modal dialog steps already. Reuse them!

File: test-cypress/steps/repository-steps.js [hunk 7]
```diff
@@ -234,4 +254,16 @@ export class RepositorySteps {
     static getLocalGraphDBTable() {
         return cy.get('#wb-locations-locationInGetLocations');
     }
+
+    static clickObdaSelectFilesBtn() {
+       cy.get('.selectFiles').first().click();
```
Comment (svilenvelikov) [inline test-cypress/steps/repository-steps.js:259]: Why `first`? Provide some context to your selectors or pass a parameter and use the `eq(index)` method in cypress

File: test-cypress/steps/repository-steps.js [hunk 8]
```diff
@@ -234,4 +254,16 @@ export class RepositorySteps {
...
+    }
+
+    static uploadObdaFile(file) {
+        cy.get('input[type=file]').first().selectFile(file, {force: true});
```
Comment (svilenvelikov) [inline test-cypress/steps/repository-steps.js:263]: Again? See above!

Comment (svilenvelikov) [inline test-cypress/steps/repository-steps.js:263]: Why `force:true`? You should always try to avoid using `force`, because it migth cover up some bug in the UI. If it's not possible to avoid it's usage, then you must document why it was needed.

File: test-cypress/integration/repository/ontop-repository.spec.js (hunk 3 above)
Comment (DesiBorisova) [reply test-cypress/integration/repository/ontop-repository.spec.js]: Fetching these parameters with `cy.fixture()` before or inside the `intercepts` caused the tests to fail. Maybe because of the file structure. In the interest of time I moved the variable to the stub file with the requests.

File: test-cypress/steps/repository-steps.js (hunk 8 above)
Comment (DesiBorisova) [reply test-cypress/steps/repository-steps.js]: Will add a comment for the need of `force` here.

File: test-cypress/steps/repository-steps.js [hunk 9]
```diff
@@ -55,6 +55,10 @@ export class RepositorySteps {
         return RepositorySteps.getRepositoryFromList(id).find('.icon-connection-off');
     }
 
+    static clickRepositoryConnectionOffBtn(id) {
```
Comment (svilenvelikov) [inline test-cypress/steps/repository-steps.js:58]: Just a side note. You don't need to add these `click`, `press`, etc. prefixes. Just write meaningful names for methods that are abstract enough, like 'createRepository`, `saveACLList`, etc. Having these prefixes, just expose implementation details in the method names and at some point they might end up out of sync with the actual implementation.

**Review by svilenvelikov — APPROVED** [review]

**Comment by DesiBorisova** [comment]
Jenkins retry build

**Comment by DesiBorisova** [comment]
Jenkins retry build
